### PR TITLE
Feature/adding settings endpoint

### PR DESCRIPTION
**Pull request by Pavlechko**
1. Created handler and service of settings
2. Created settings router
3. Added Background settings to settings handler and service
4. Added Quote settings to settings handler and service
5. Added Weather settings to settings handler and service
6. Added Exchange settings to settings handler and service
7. Added Market settings to settings handler and service
8. Added returning settings in home endpoint

### CODE REVIEW

File: internal/models/weather.model.go [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
 
-type WeatherData struct {
-	OpenWeather     FrontendWeatherResponse
-	TomorrowWeather FrontendWeatherResponse
+type WeatherInput struct {
+	City   string `json:"city" binding:"required"`
+	Source string `json:"source" binding:"required"`
 }
 
+// type WeatherData struct {
```
Comment (cx-serhii-stanislavskyi) [inline internal/models/weather.model.go]: If you don't need it better to take it out

File: internal/services/background.service.go [hunk 2]
```diff
@@ -110,3 +97,17 @@ func getUnsplashBackgroundImage() models.FrontendBackgroundImageResponse {
...
+func GetRandomBackground(source string) models.FrontendBackgroundImageResponse {
+	if source == "unsplash.com" {
+		return getUnsplashBackgroundImage()
+	}
+	return getPexelsBackgroundImage()
+}
+
+func GetBackgroundData() models.FrontendBackgroundImageResponse {
+	// TO-DO
+	// Check in the DB how long ago the image was updated and return the new one or retrieve the old one from the DB
```
Comment (cx-serhii-stanislavskyi) [inline internal/services/background.service.go]: Better write it in such way: "TODO: Check in...."

File: internal/services/exchange.service.go [hunk 3]
```diff
@@ -8,81 +8,85 @@ import (
...
 
 	_ "github.com/joho/godotenv/autoload"
 )
 
 var currentTime = time.Now()
 var yesterday = currentTime.AddDate(0, 0, -1)
-var symbolsArr = []string{"AUD", "BRL", "BTC", "CAD", "CLP", "CNY", "CZK", "EGP", "EUR", "GBP", "HKD", "INR", "JPY", "KRW", "LTL", "LVL", "TRY", "USD", "XAG", "XAU"}
 
-func getNBUData(date string) []models.NBU {
+// var symbolsArr = []string{"AUD", "BRL", "EGP", "CAD", "CLP", "CNY", "CZK", "EGP", "EUR", "GBP", "HKD", "INR", "JPY", "KRW", "LTL", "LVL", "TRY", "USD", "XAG", "XAU", "UAH", "PLN"}
```
Comment (cx-serhii-stanislavskyi) [inline internal/services/exchange.service.go]: Better write it in such way: Available currencies: []string{ "AUD", ....

File: internal/services/exchange.service.go [hunk 4]
```diff
@@ -8,81 +8,85 @@ import (
...
 
-	apiURL := fmt.Sprintf("https://bank.gov.ua/NBUStatService/v1/statdirectory/exchange?date=%s&json", date)
+	apiURL := fmt.Sprintf("https://bank.gov.ua/NBUStatService/v1/statdirectory/exchange?valcode=%s&date=%s&json", symbol, date)
 
 	resp, err := http.Get(apiURL)
 
 	if err != nil {
 		utils.ErrorLogger.Println("Error creating HTTP request:", err)
-		return response
+		return response[0]
```
Comment (cx-serhii-stanislavskyi) [inline internal/services/exchange.service.go]: Check it before if array isn't empty

File: internal/services/weather.services.go [hunk 5]
```diff
@@ -13,17 +13,12 @@ import (
 	_ "github.com/joho/godotenv/autoload"
 )
 
-func GetWeatherData() models.WeatherData {
+func GetWeatherData() models.FrontendWeatherResponse {
+	// make request to DB
+	openWeatherRes := getOpenWeatherData("kyiv")
+	// tomorrowWeatherRes := getTomorrowWeatherData("kyiv")
```
Comment (cx-serhii-stanislavskyi) [inline internal/services/weather.services.go]: Better remove it

File: internal/utils/token.go [hunk 6]
```diff
@@ -31,3 +33,35 @@ func GenerateJWT(user *models.UserResponse) (string, error) {
...
+
+func GetUserId(r *http.Request) string {
+	if r.Header["Authorization"] != nil {
+
+		authHeader := strings.Split(r.Header.Get("Authorization"), "Bearer ")
+
+		if len(authHeader) != 2 {
+			ErrorLogger.Println("Malformed Token")
+			return ""
+		} else {
```
Comment (cx-serhii-stanislavskyi) [inline internal/utils/token.go]: You can write it without else. You have return in previous block

**Review by cx-serhii-stanislavskyi — CHANGES_REQUESTED** [review]

File: internal/models/weather.model.go (hunk 1 above)
Comment (Pavlechko) [reply internal/models/weather.model.go]: Fixed: Deleted this comment

File: internal/services/background.service.go (hunk 2 above)
Comment (Pavlechko) [reply internal/services/background.service.go]: Fixed: Renamed this comment

File: internal/services/exchange.service.go (hunk 3 above)
Comment (Pavlechko) [reply internal/services/exchange.service.go]: Fixed: Deleted this comment

File: internal/services/exchange.service.go (hunk 4 above)
Comment (Pavlechko) [reply internal/services/exchange.service.go]: Fixed: Default structure is now returned on error. There is also a check an empty response.

File: internal/services/weather.services.go (hunk 5 above)
Comment (Pavlechko) [reply internal/services/weather.services.go]: Fixed: Deleted these comments

File: internal/utils/token.go (hunk 6 above)
Comment (Pavlechko) [reply internal/utils/token.go]: Fixed: Rewritten without else-block

**Review by cx-serhii-stanislavskyi — APPROVED** [review]

**Pull request merged**